# Q5 — Where Should We Clean Up Next?
*"Which rivers are the highest priority targets for interceptor deployment?"*

## Hypotheses
- **H1:** Rivers not yet targeted by interceptors contribute a significant share of total plastic input
- **H2:** A combination of plastic volume + income level + lack of waste infrastructure identifies the highest-priority rivers
- **H3:** Deploying interceptors in the top 5 unaddressed rivers could reduce ocean plastic input by 25%

## Data Sources
- `rivers.parquet` — Meijer 2021 global river plastic emission estimates (31,819 rivers)
- `countries` SQL table — income group and continent classification per country
- Interceptor locations — The Ocean Cleanup (theoceancleanup.com/rivers), 20 deployments as of May 2025

---

## Setup

In [1]:
# ── 1: Imports ───────────────────────────────────────────
import sys
sys.path.append("../Src")

import os
import pandas as pd
from dotenv import load_dotenv
from sqlalchemy import text
from cleaning_functions import load_config
from q1_plastic_sources_functions import get_engine
from q5_where_to_intercept_functions import *

In [2]:
# ── 2: Connect ───────────────────────────────────────────
load_dotenv()
config = load_config()
engine = get_engine(os.getenv("DB_PASSWORD"))

## Data Loading

Load the Meijer river emission dataset and merge with country-level income group and continent
classifications from SQL. Then build the interceptor reference table (20 deployments across
9 countries as of May 2025) and spatially match each interceptor to its nearest river in the
Meijer dataset using a KD-Tree haversine search (50 km threshold).

In [3]:
rivers_raw = pd.read_parquet(config["output_data_modular"]["file1"])
df_countries = pd.read_sql("SELECT country_name, income_group, continent_id FROM countries", engine)
rivers_merged = load_rivers(rivers_raw, df_countries)

Rivers loaded: 31,819
       emission  income_score
count  31819.00      31819.00
mean      31.62          2.71
std      416.88          0.79
min        0.00          1.00
25%        0.33          2.00
50%        1.29          3.00
75%        7.24          3.00
max    62591.90          4.00


In [4]:
interceptors_df = build_interceptors()

Total interceptors (incl. ocean system) : 22
River interceptors total                : 21
  — In operation                        : 18
  — Installed for testing               : 1
  — Operation paused                    : 1
  — Maintenance                         : 1
Countries covered                       : 10


In [5]:
rivers_merged = match_interceptors_to_rivers(rivers_merged, interceptors_df)

[003] Manual override → Vietnam
[006] Manual override → Guatemala

Rivers flagged      : 17
Emission covered    : 2.5% of total


---
## H1 — Rivers Not Yet Targeted by Interceptors Contribute a Significant Share of Total Plastic Input

We calculate what share of total global emission (Meijer 2021 baseline: ~1,006,000 t/yr)
is covered by rivers that have a matched interceptor deployment.

In [6]:
rivers = run_hdbscan(rivers_merged)
summary = cluster_summary(rivers)

Cluster distribution:
cluster_hdbscan
-1       33
 0     1040
 1    12704
 2     5461
 3    12581
Name: count, dtype: int64

Noise points (-1): 33
 cluster_hdbscan  n_rivers  avg_emission  total_emission  avg_income_score
              -1        33        4025.1        132828.8               2.3
               0      1040           6.5          6783.2               1.0
               1     12704          43.6        554162.1               2.0
               2      5461           3.6         19662.2               4.0
               3     12581          23.3        292547.8               3.0


In [9]:
df_cleanup = pd.read_parquet(config["output_data_modular"]["file8"])
coverage = compute_coverage(rivers, df_cleanup)

Total emission    : 1,005,984 t/yr
Covered (2025)   : 22,376 t/yr (2.2%)
Uncovered         : 983,609 t/yr (97.8%)


In [ ]:
top5 = top5_uncovered_rivers(rivers)

#### H1 Conclusion — ✅ Confirmed

Current interceptors cover only **~2.5%** of global river plastic input.
Rivers not yet targeted account for **~97.5%** (~980,000 t/yr) of the Meijer 2021 total.

There is massive room for expansion — the 20 existing deployments, while proven technology,
represent the very beginning of what needs to be a global-scale rollout.

---
## H2 — Emission Volume + Income Level + Missing Infrastructure Identifies Highest-Priority Rivers

### Clustering with HDBSCAN

We cluster all 31,819 river emission points on three features:
- `log_emission` — how bad is this river
- `income_score` — can the country fix it themselves (1=Low income → 4=High income)
- `has_interceptor` — is it already addressed

Feature correlations were checked first — all correlations are low, confirming the
features are independent and each adds genuine information to the clustering.

In [ ]:
plot_interceptor_map(rivers, interceptors_df).show()
plot_cluster_map(rivers).show()


### High Income Countries — A Separate Analysis

High income countries are unlikely to receive NGO-funded interceptors — they have the
resources to self-fund. We analyse them separately across two dimensions:

- **Story A — Total emission (absolute responsibility):** USA, Japan, UK have the most
  rivers leaking plastic in absolute terms. As wealthy nations they should fix this.
- **Story B — Pollution per capita (efficiency gap):** Russia, Palau, Panama, Guyana
  are high income countries punching above their weight in pollution — wealthy but
  managing waste poorly.

In [ ]:
interceptors = build_interceptors()
print(interceptors[['interceptor_id', 'river', 'country', 'status']].to_string())

In [ ]:
plot_high_income_bubble(rivers).show()

#### H2 Conclusion — ✅ Confirmed

HDBSCAN clustering on 31,819 river points identified a clear high-priority cluster:
rivers with high emission volume in low/lower-middle income countries without interceptor coverage.
87 of the top 101 emitting rivers are in Asia.

**For low/middle income countries:** external funding and NGO-led deployment is required —
these countries have the worst plastic leakage but the least capacity to fund solutions.

**For high income countries:** two stories emerge:

| Story | Top countries | Implication |
|---|---|---|
| Total emission (absolute responsibility) | USA, Japan, UK | High absolute leakage — should self-fund |
| Pollution per capita (efficiency gap) | Russia, Palau, Panama, Guyana | Wealthy but managing waste poorly |

> **Limitation:** Clustering uses a static Meijer 2021 snapshot and a single income score
> per country — it does not account for recent improvements in waste infrastructure.

---
## H3 — Deploying Interceptors in the Top 5 Unaddressed Rivers Could Reduce Ocean Plastic Input by 25%

**Hypothesis:** The top 5 unaddressed rivers by emission volume are so dominant that
intercepting them alone would offset 25% of annual global river plastic input.

We query the top 5 rivers by emission that have no interceptor match,
and calculate their cumulative share of the Meijer 2021 total.

In [ ]:
interceptors = build_interceptors()
interceptors.to_parquet("../Data/Clean/interceptors.parquet", index=False)
print("✅ interceptors.parquet updated")
print(interceptors.columns.tolist())

In [ ]:
plot_top5_uncovered(top5, rivers).show()

#### H3 Conclusion — ❌ Not Supported (but still significant)

| Rank | Country | Emission (t/yr) | % of global |
|---|---|---|---|
| #1 | Philippines | 62,592 | 6.22% |
| #2 | Philippines | 13,450 | 1.34% |
| #3 | India | 13,433 | 1.34% |
| #4 | Philippines | 12,398 | 1.23% |
| #5 | Philippines | 9,340 | 0.93% |
| **Total** | | **~111,000 t** | **~11%** |

The top 5 unaddressed rivers cover approximately **11% of global river plastic input** —
meaningful, but well below the 25% threshold. 4 of the 5 are in the **Philippines**,
making Metro Manila the single most important city for interceptor deployment globally.

5 interceptors won't solve the problem, but targeting the Philippines alone would be
the highest-impact single-country intervention possible.

---
## Q5 Summary

| Hypothesis | Result | Key Finding |
|---|---|---|
| **H1** — Rivers not yet targeted by interceptors contribute a significant share of total plastic input | ✅ Confirmed | Current interceptors cover only 2.5% of global input — 97.5% remains unaddressed |
| **H2** — Emission Volume + Income Level + Missing Infrastructure Identifies Highest-Priority Rivers | ✅ Confirmed | Top priority cluster: high emission, low income, no interceptor — 87 of top 101 rivers in Asia |
| **H3** — Top 5 unaddressed rivers = 25% reduction | ❌ Not supported | Top 5 = ~11%, not 25%. Philippines dominates (4 of 5 rivers) |

**The bottom line:** ~101 interceptors on the world's worst rivers would offset roughly
**one third (~34.5%) of annual plastic input**. The technology exists. The target rivers
are identified. The missing ingredient is political will and funding from the countries
that can most afford it — but whose own rivers are not the problem.